src/charm.py

```diff
@@ -136,6 +143,39 @@ def _on_config_changed(self, event: ops.framework.EventBase) -> None:
 
             self.configure_charm(event)
 
+    def _handle_receive_ca_cert(self, event: ops.framework.EventBase) -> None:
```

Comment:
can we move this method outside of `charm.py` ? This would be really helpful in organising the code.

Reply:
As discussed, i will be leaving this function in charm.py